SIPS/sip-3.md

```diff
@@ -93,63 +98,71 @@ interface OnTransactionArgs {
 
 ```typescript
-interface LegacyTransactionObject {
+interface OnTransactionReturn {
+  insights: Record<string, Json>;
+}
+```
+
+### MetaMask Integration
+
+The `insights` object returned by the snap will be displayed alongside the confirmation for the `transaction` that `onTransaction` was called with.
+Keys and values will be displayed in the order received, with each key rendered as a title and each value rendered as follows:
+
+- If the value is an array or an object, it will be rendered as text after being converted to a string.
+- If the value is neither an array nor an object, it will be rendered directly as text.
+
+## Appendix I: Ethereum Transaction Objects
+
+The following transaction objects may appear for any `chainId` of `eip155:*` where `*` is some positive integer.
+This includes all Ethereum or "EVM-compatible" chains.
+As of the time of creation of this SIP, they are the only possible transaction objects for Ethereum chains.
+
+### EIP-1559
+
+```typescript
+interface TransactionObject {
   from: string;
   to: string;
   nonce: string;
   value: string;
   data: string;
   gas: string;
-  gasPrice: string;
+  maxFeePerGas: string;
+  maxPriorityFeePerGas: string;
   type: string;
   estimateSuggested: string;
   estimateUsed: string;
 }
 ```
 
-_[EIP-1559]_
+### Legacy (non-EIP-1559)
 
 ```typescript
-interface TransactionObject {
+interface LegacyTransactionObject {
   from: string;
   to: string;
   nonce: string;
   value: string;
   data: string;
   gas: string;
-  maxFeePerGas: string;
-  maxPriorityFeePerGas: string;
+  gasPrice: string;
   type: string;
   estimateSuggested: string;
   estimateUsed: string;
 }
 ```
 
-**ChainId** - This is a [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) `chainId` string, the snap is expected to parse and utilize this string as needed.
-
-The return type for an `onTransaction` export should be as follows:
-
-```typescript
-interface OnTransactionReturn {
-  insights: Record<string, Json>;
-}
-```
-
-### MetaMask Extension Integration
-
-A transaction insight snap (a snap with the `endowment:transaction-insight` permission) will be displayed in an extra tab in MetaMask confirmation screens. Currently, transaction insights will be provided for contract interaction type transactions. The `insights` object returned from the snap will be displayed in the confirmation screen UI as titles and subtext. The key being the title and the subtext being a stringified version of the value.
-
-**Note:** In the future, the intention is to extend the `OnTransactionReturn` interface with additional properties that can interact with the extension in various ways.
-
-## History
-
-The transaction insight feature has been inspired by the need to improve user confidence in everyday web3 interactions. Allowing for various transaction insights inside of MetaMask will increase confidence, improve UX and ultimately provide a safer and more informed experience for our users.
-
```

Comment:
This kind of specification metadata should not go in the specification itself, in the interest of keeping it succinct.